akka-actor/src/main/scala/akka/util/WildcardIndex.scala

```diff
@@ -53,18 +54,31 @@ private[akka] final case class WildcardTree[T](
       copy(data = Some(d))
     } else {
       val e = elems.next()
-      copy(children = children.updated(e, children.getOrElse(e, WildcardTree[T]()).insert(elems, d)))
+      if (e != "*" && e != "**" && e.endsWith("*"))
+        copy(
+          wildcardSuffixChildren = wildcardSuffixChildren
+            .updated(e.stripSuffix("*"), wildcardSuffixChildren.getOrElse(e, WildcardTree[T]()).insert(elems, d)))
```

Comment:
Should it perhaps fail for `e != "**" && e.endsWith("**")` for extra clarity that doesn't work?